### PR TITLE
feat: cli and init command, text-field component

### PR DESCRIPTION
**Pull request by arihanv**
- Created the rubricui cli with an init command to create the preset, CSS, and tailwind config file
- Created the text-field (input) component

### CODE REVIEW

File: bin/config/rubric.preset.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const rubricConfig = {
+	content: ["./node_modules/rubricui/src/**/*.{js,jsx,ts,tsx}"],
+	theme: {
+		extend: {
+			colors: {
+				"brand-primary": {
```
Comment (sarimrmalik) [inline bin/config/rubric.preset.js]: If we are using the values in globals.css, this would then be `"brand-primary": "var(--primary)"` → if we use hex codes in the globals.css file. For HSL, I think you need to add a prefix with the variable name. 

So that we don't duplicate information.

Comment (sarimrmalik) [reply bin/config/rubric.preset.js]: I see that the `rubric.preset.ts` is correct, I think the .js and .ts files need to be matched.

Comment (arihanv) [reply bin/config/rubric.preset.js]: Fixed, made it the same as `rubric.preset.ts` which was correct

File: src/components/input.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	size?: keyof typeof sizes;
+}
+
+const Input = forwardRef<HTMLInputElement, InputProps>(
+	({ className, size = "small", type, ...props }, ref) => {
+		return (
+			<input
+				type={type}
+				className={cn(
+					"flex items-center self-stretch placeholder:text-gray-400 text-xs leading-[130%] tracking-[0.12px] border bg-contrast/5 border-contrast/10 focus:outline-contrast/20",
```
Comment (sarimrmalik) [inline src/components/input.tsx]: `text-gray-400` → would this be a configurable variable?

Let's say we had a `brand-foreground` variable which had the text color and we could use placeholder:text-brand-foreground/50 to get that same effect so the brand stays consistent and is configurable. Thoughts?

File: tailwind.config.ts [hunk 3]
```diff
@@ -1,16 +1,12 @@
...
     extend: {
       colors: {
-        "brand-primary": {
-          DEFAULT: "#FCFCFC",
-        },
-        "brand-contrast": {
-          DEFAULT: "#050505",
-        },
+        primary: "rgb(var(--primary) / <alpha-value>)",
+        contrast: "rgb(var(--contrast) / <alpha-value>)",
```
Comment (sarimrmalik) [inline tailwind.config.ts]: Why the change?

Comment (arihanv) [reply tailwind.config.ts]: This file doesn't actually have an effect on the component styling since only the field from `bin/config` are injected

File: src/components/input.tsx (hunk 2 above)
Comment (arihanv) [reply src/components/input.tsx]: What would the foreground color be? Should we define a new color for foreground or could we instead use `text-contrast/50`?

File: tailwind.config.ts (hunk 3 above)
Comment (sarimrmalik) [reply tailwind.config.ts]: I see, but if there are not benefits of this naming, let's use the default naming to avoid confusion

File: src/components/input.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	size?: keyof typeof sizes;
+}
+
+const Input = forwardRef<HTMLInputElement, InputProps>(
+	({ className, size = "small", type, ...props }, ref) => {
+		return (
+			<input
+				type={type}
+				className={cn(
+					"flex items-center self-stretch placeholder:text-gray-400 border bg-contrast/5 border-contrast/10 focus:outline-contrast/20",
```
Comment (sarimrmalik) [inline src/components/input.tsx]: I left a comment on the `text-gray-400`, would love clarity on that and use a global text color with a lower opacity here for consistency & configurability.

Comment (arihanv) [reply src/components/input.tsx]: Replied to that:
> What would the foreground color be? Should we define a new color for foreground or could we instead use `text-contrast/50`?

**Review by sarimrmalik — APPROVED** [review]
LGTM 🫡
